### DISCUSSION TITLE
Distribution of Exogenous Contracts

### DISCUSSION THREAD

**Discussion by dipplestix** [yasserfarouk/scml#28] [one-shot-qa] [answered]
The paper says
" These totals are divided among the corresponding factories randomly, but in such away that the fraction of exogenous contracts assigned to one factory relative to others at the same level remains consistent throughout a simulation."

Does this mean each factory will get around the same amount of exogenous contracts throughout the game (~1/3 if there are three factories) or will they receive different amounts (.6, .3, .1 for instance)?

**Comment by yasserfarouk** [comment, marked as answer]
The total number of exogenous contracts varies per step in general. Say it was 20, 25. 15, 30

Each factory gets a similar but not exactly equal share of these. For factory A this may be .3, .25, .35, .3

As a result, the factory will get quantities of around:
20*.3~=7, 25*.25~=6, 15*.35~=7, 30*.3~=10

So as you see both fractions and absolute quantities vary per factory but the variation is not large.

**Reply by dipplestix** [reply]
Are these sampled from some distribution (Like N(1/F_l, small)) or do they live inside the world/AWI information somewhere?

**Reply by yasserfarouk** [reply]
They are sampled while generating the world configuration from one distribution per agent (so agents can have different variances not only means) and corrected to make sure that all quantities sum to the total exogenous quantity per day. These are stored in the world and are not accessible by agents. Every day, the exogenous contract of each agent is revealed to it alone and the total quantity/average price is revealed in the bulletin board.

You can see the code [here](https://github.com/yasserfarouk/scml/blob/master/src/scml/common.py) in `distribute_quantities`. We do not guarantee that this will be the same code used in the final competition though.

**Reply by yasserfarouk** [reply]
@dipplestix  Please let me know if your question is answered. Thanks
